From 63c4dc41d2b90dc3386e331b62ee893a32bb0b11 Mon Sep 17 00:00:00 2001
From: Maksim Sisov <msisov@igalia.com>
Date: Wed, 16 Aug 2017 10:56:58 +0000
Subject: [PATCH] Fix compilation failures for arm64.

This cl fixes some compilation failures we faced when compiling
for arm64 (aarch64, basically, with gcc). As long as
official chrome is also compiled for arm64, we propose a fix
for these trivial problems:

../../ui/views/examples/box_layout_example.cc: In member function 'void views::examples::{anonymous}::ChildPanel::SetSelected(bool)':
../../ui/views/examples/box_layout_example.cc:167:34: error: 'void views::examples::BoxLayoutExample::RefreshLayoutPanel()' is private within this context
  example_->RefreshLayoutPanel();
                               ^

and

../../chrome/browser/devtools/devtools_file_system_indexer.cc:37:12: error: 'set' is already declared in this scope
  using std::set;
               ^

Bug: None
Change-Id: I08cc6e0bc3e5819ab4f13f39a9abf5bd4f1dd1a5
Reviewed-on: https://chromium-review.googlesource.com/612984
Commit-Queue: Maksim Sisov <msisov@igalia.com>
Reviewed-by: Dmitry Gozman <dgozman@chromium.org>
Reviewed-by: Sadrul Chowdhury <sadrul@chromium.org>
Cr-Commit-Position: refs/heads/master@{#494745}
---
 chrome/browser/devtools/devtools_file_system_indexer.cc | 3 ++-
 ui/views/examples/box_layout_example.h                  | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/chrome/browser/devtools/devtools_file_system_indexer.cc b/chrome/browser/devtools/devtools_file_system_indexer.cc
index 87efb803918d..576a620649dd 100644
--- a/chrome/browser/devtools/devtools_file_system_indexer.cc
+++ b/chrome/browser/devtools/devtools_file_system_indexer.cc
@@ -34,12 +34,13 @@ using base::TimeDelta;
 using base::TimeTicks;
 using content::BrowserThread;
 using std::map;
-using std::set;
 using std::string;
 using std::vector;
 
 namespace {
 
+using std::set;
+
 base::SequencedTaskRunner* impl_task_runner() {
   constexpr base::TaskTraits kBlockingTraits = {base::MayBlock(),
                                                 base::TaskPriority::BACKGROUND};
diff --git a/ui/views/examples/box_layout_example.h b/ui/views/examples/box_layout_example.h
index 76950459ecee..d6b0a2cf2b5a 100644
--- a/ui/views/examples/box_layout_example.h
+++ b/ui/views/examples/box_layout_example.h
@@ -38,7 +38,7 @@ class VIEWS_EXAMPLES_EXPORT BoxLayoutExample : public ExampleBase,
   void CreateExampleView(View* container) override;
 
  private:
-  friend class ChildPanel;
+  friend class views::examples::ChildPanel;
   // ButtonListener
   void ButtonPressed(Button* sender, const ui::Event& event) override;
 
-- 
2.11.0

